projects/fsastorefrontlib/src/cms-components/checkout/checkout.module.ts

```diff
@@ -212,6 +213,7 @@ const routes: Routes = [
         AddOptionsFlex: {
           // mapping hybris component (defined in impex) - This is acctualy flexType defined in impex for that component
           component: AddOptionsComponent, // to SPA component
+          guards: [AddOptionsGuard],
```

Comment:
Naming could be improved. This is essentially guard that can be injected at any page and it will load/merge provided cart into current active cart and it's associated to chatbot feature.
Maybe something like ChatbotActiveCartGuard